src/plugins/intel_cpu/tests/functional/utils/cpu_test_utils.cpp

```diff
@@ -471,4 +472,43 @@ void CheckNumberOfNodesWithType(const ov::CompiledModel& compiledModel,
                                 size_t expectedCount) {
     CheckNumberOfNodesWithTypes(compiledModel, {nodeType}, expectedCount);
 }
+
+
+ov::element::Type
+CPUTestsBase::get_default_imp_precision_type(const ov::element::Type& in_type,
+                                             const ov::AnyMap& configuration) {
+#if defined(OPENVINO_ARCH_ARM) || defined(OPENVINO_ARCH_ARM64)
+    return in_type;
+#endif
+#if defined(OPENVINO_ARCH_RISCV64)
+    return in_type;
+#endif
+#if defined(OPENVINO_ARCH_X86_64)
+    const std::string key = ov::hint::inference_precision.name();
+    // if is not float
+    if (!in_type.is_real()) {
+        return in_type;
+    }
+
+    ov::element::Type type = in_type;
+    // ngraph tranform stage
+    if (type == ov::element::bf16) {
+        type = ov::with_cpu_x86_avx512_core() ? ov::element::bf16 : ov::element::f32;
+    } else {
+        type = ov::element::f32;
+    }
```

Comment:
This behavior isn't that clear for the users of this call. Actually the type is preserved only when the inference precision in the config is "undefined", in other scenarios the  expected type is determined based on the inference precision and the HW capabilities. Shouldn't we align the behavior of this utility call with the behavior of the plugin?

Reply:
Currently from my knowledge about the CPU plugin code, there are two stages to confirm the precision,
1. bf16=>fp32 and fp16=>fp32 (transform stage) https://github.com/openvinotoolkit/openvino/blob/63a13d4931c829172388a43a44785eda7b45717d/src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp#L351
2. enforce precision (configure stage)
https://github.com/openvinotoolkit/openvino/blob/63a13d4931c829172388a43a44785eda7b45717d/src/plugins/intel_cpu/src/graph.cpp#L1753

Reply:
because https://github.com/openvinotoolkit/openvino/pull/23950 will be merged into master ,the behavior of plugin about precision fp16 will change, will update this function to align with new behavior

Reply:
already updated